deepface/modules/verification.py

```diff
@@ -188,21 +156,18 @@ def verify(
         "right_eye": None,
     }
 
-    distances = []
-    facial_areas = []
+    min_distance, min_idx, min_idy = float("inf"), None, None
     for idx, img1_embedding in enumerate(img1_embeddings):
         for idy, img2_embedding in enumerate(img2_embeddings):
             distance = find_distance(img1_embedding, img2_embedding, distance_metric)
-            distances.append(distance)
-            facial_areas.append(
-                (img1_facial_areas[idx] or no_facial_area, img2_facial_areas[idy] or no_facial_area)
-            )
+            if distance < min_distance:
+                min_distance, min_idx, min_idy = distance, idx, idy
 
     # find the face pair with minimum distance
     threshold = threshold or find_threshold(model_name, distance_metric)
-    min_index = np.argmin(distances)
-    distance = float(distances[min_index])  # best distance
-    facial_areas = facial_areas[min_index]
+    distance = float(min_distance)
+    facial_areas = (no_facial_area, no_facial_area) if None in (min_idx, min_idy) else \
```

Comment:
what if img1_facial_areas[min_idx] is None and img2_facial_areas[min_idy] is not None?

In previous design, we were setting no_facial_area's values to 1st item, and img2_facial_areas[min_idy]'s value to 2nd item.

In your design, no_facial_area's values are not set, instead you are setting None. Am I right?

Reply:
I think the following code snippet will resolve that issue

```python
facial_areas = (
	no_facial_area if min_idx is None else img1_facial_areas[min_idx] or no_facial_area,
	no_facial_area if min_idy is None else img2_facial_areas[min_idy] or no_facial_area,
)
```

Reply:
As I understand it, `img1_facial_areas[min_idx]` and `img2_facial_areas[min_idy]` will never be `None`, because if a face is not found, the area of the entire image is still returned. Therefore, the code from the previous design, `img1_facial_areas[idx] or no_facial_area`, will never return `no_facial_area`.

That's why I thought that if we don't find faces in at least one of the photos, we shouldn't display anything at all.

The code you suggested really solves the problem, thank you!

Reply:
Maybe we don't need `or no_facial_area` in the end of statements? Because `no_facial_area if min_idx is None else img1_facial_areas[min_idx]` will return `Truthy` anyway.
```Python
facial_areas = (
	no_facial_area if min_idx is None else img1_facial_areas[min_idx],
	no_facial_area if min_idy is None else img2_facial_areas[min_idy],
)
```

Reply:
Actually, img1_facial_areas[idx] can be None from this line

https://github.com/serengil/deepface/blob/master/deepface/modules/verification.py#L166

Reply:
No, the problem occurs when min_idx is 0 but img1_facial_areas[0] is None. We still need to return the dictionary, but the code snippet you recommended returns None.

Reply:
Yes, you are right. I didn't mention, that `img1_facial_areas[idx]` can return `None`. Sorry, my bad.

What do you think, maybe, it is better to do something like this?
```python
img_facial_areas = [no_facial_area]
```
instead of 
```python
img_facial_areas = [None]
```

Reply:
yeah, this will also work

i hope this is the only point facial area can be none - https://github.com/serengil/deepface/blob/master/deepface/modules/verification.py#L166